### PR TITLE
Add highest L1 and L2 block number Gauge metrics to DTL

### PR DESCRIPTION
**Pull request by snario**
Helps with debugging and evaluating sync progress.

### CODE REVIEW

**Review by annieke — DISMISSED** [review]
this looks good! this should also preserve our express http metrics along with creating the new ones on sync height

File: packages/data-transport-layer/src/services/l2-ingestion/service.ts [hunk 1]
```diff
@@ -109,6 +126,8 @@ export class L2IngestionService extends BaseService<L2IngestionServiceOptions> {
 
         await this.state.db.setHighestSyncedUnconfirmedBlock(targetL2Block)
 
+        this.l2IngestionMetrics.highestSyncedL2Block.set(targetL2Block)
```
Comment (tynes) [inline packages/data-transport-layer/src/services/l2-ingestion/service.ts:129]: Are there problems here if `this.metrics` is `false`? Will it result in a `cannot call set on undefined` error?

Comment (annieke) [reply packages/data-transport-layer/src/services/l2-ingestion/service.ts:129]: ^ohh this is a good catch, i think it probably would

Comment (snario) [reply packages/data-transport-layer/src/services/l2-ingestion/service.ts:127]: Changed it to make `metrics` mandatory

**Comment by tynes** [comment]
Could you add a changeset to this?

**Comment by tynes** [comment]
Does this need to run a `metricsServer`?

https://github.com/ethereum-optimism/optimism/blob/3b60ea63f8cfffac9b9f84f997f43837f84ee92f/packages/batch-submitter/src/exec/run-batch-submitter.ts#L482-L490

**Comment by snario** [comment]
[`express-prom-bundle`](https://github.com/jochen-schweizer/express-prom-bundle) is meant to be serving metrics as a middleware.

**Comment by tynes** [comment]
This needs a changeset still

**Pull request merged**